src/runtime/entitlements-manager.js

```diff
@@ -50,16 +51,61 @@ import {warn} from '../utils/log';
 
 const SERVICE_ID = 'subscribe.google.com';
 
+/**
+ * Properties:
+ *   - isClosable - determine whether the view is closable.
+ *   - onResult - callback to get the intervention result and decide if it completes.
+ *                Takes either a normal or async function and returns `true` if the
+ *                intervention should be marked complete.
+ *
+ * @typedef {{
+ *   isClosable: (boolean|undefined),
+ *   onResult: ((function(!Object):(Promise<Boolean>|Boolean))|undefined),
+ * }}
+ */
+export let ShowInterventionParams;
+
+class Intervention {
+  /** @public @const {string} */
+  type;
+  /** @public @const {string} */
+  configurationId;
+}
+
+export class AvailableIntervention extends Intervention {
+  /**
+   * @param {Intervention} original
+   * @param {!./deps.DepsDef} deps
+   */
+  constructor(original, deps) {
+    super();
+    Object.assign(this, original);
+    /** @private @const {!./deps.DepsDef} */
+    this.deps_ = deps;
+  }
+  /**
+   * Starts the intervention flow.
+   * @param {!ShowInterventionParams=} params
+   * @return {!Promise}
+   */
+  show(params) {
+    // TODO: implement `onResult`
+    const flow = new AudienceActionFlow(this.deps_, {
+      isClosable: params.isClosable,
+      action: this.type,
+    });
+    return flow.start();
+  }
+}
+
 /**
  * Article response object.
  *
  * @typedef {{
  *  entitlements: (../api/entitlements.Entitlements),
  *  clientConfig: (../model/client-config.ClientConfig),
  *  audienceActions: ({
- *    actions: Array<{
- *      type: (string)
- *    }>,
+ *    actions: Array<Intervention>,
```

Comment:
Just checking, can the entries be `null`?
```suggestion
 *    actions: Array<!Intervention>,
```